### DISCUSSION TITLE
Icaros 3.3.3 Beta 3

### DISCUSSION THREAD

**Discussion by Xanashi** [Xanashi/Icaros#132] [Releases]
Good Afternoon Everyone,

This should be the last beta release, before I release 3.3.3 final, which I hope to do soon. 
It mostly just contains bug fixes and small improvements. 

Let me know if you find any regressions or new bugs. 

I hope you all enjoy the new release! ❤️ 

<br/>

[![Icaros releases](https://img.shields.io/github/downloads/xanashi/icaros/v3.3.3b3/total?color=%232B6870&label=DOWNLOADS&style=for-the-badge)](https://github.com/Xanashi/Icaros/releases/tag/v3.3.3b3)

_______________________________________________________________________________________________________

### Icaros v.3.3.3 Beta 3

- Implemented new INDD file parser (should fix previous INDD issues)
- Fixed thumbnailing of several HEVC files
- Fixed MP4 'creation date' property being off by 1 hour
- Fixed 'tiled' HEIC aspect ratio issue
- Added orientation support for 'tiled' HEICs
- Added thumbnailing support for webp in ID3v2 tags
- Added AVIF, PX, EXR and PDXPRJ to the presets
- Register PDF, SVG, EXR, HEIC, HEIF, AVIF and PSXPRJ as image formats
- Updated FFmpeg

<hr /><em>This discussion was created from the release <a href='https://github.com/Xanashi/Icaros/releases/tag/v3.3.3b3'>Icaros 3.3.3 Beta 3</a>.</em>

**Comment by Qhqmvga7BWJs39sHZ** [comment]
FYI, during install, the setup dialog still says Icaros 3.3.2 and Programs and Features shows version 3.3.2.303.  The Icaros config About tab shows the correct version.

**Reply by Xanashi** [reply]
I'm aware. It's just me being lazy.
I never bothered to write the installer code that would show the correct version in the installer. 
I always felt like my time would be better spend on other features, since it's just a minor cosmetic issue only testers will encounter.
I don't believe I can do anything about the Programs and Features version, although I haven't checked honestly.

**Comment by 2297640752** [comment]
Thank you very much for the software you developed, it helped me a lot. But I found some small problems in the process of using it, when I have a lot of images, the thumbnail cache file takes up too much hard disk space, can you add the ability to manually adjust the thumbnail definition or pixel size to ensure that the clarity of the thumbnail meets the requirements and does not take up too much hard disk space.
Finally, thank you again for the software you shared.

**Reply by Xanashi** [reply]
Thank you for the kind words. I really appreciate it! <3

Unfortunately Explorer determines the sizes of the thumbnails, and since the release Windows 10 the most commonly used thumbnail size Explorer requests is 1280x1280, which is not a small thumbnail. 
In comparison, the most commonly used thumbnail size in Windows 7 was 96x96. 

This of course means, that even with compression, the thumbnail cache in later versions of Windows just takes up a lot more space. 

I'm not sure what happens in the Windows cache, if I sent back forced smaller thumbnails from Icaros. 
It may still render correctly in Explorer, while lessening the size of the cache. I could experiment a bit with it, 
and provide an option for users with limited hdd space. However it may not work.

**Comment by clsid2** [comment]
Perhaps store in compressed format in cache? Lossless PNG can give 2x reduction. Lossy webp up to 5x.

Downside that it comes at a performance cost when generating thumbs and when accessing a thumb from cache.

**Reply by Xanashi** [reply]
I believe @2297640752 was talking about the Windows Cache. The Icaros Cache already utilizes compression on all the larger thumbnails stored. It does come with a small performance cost, but with 1280x1280 thumbnails, it's very necessary.

**Comment by amayra** [comment]
> Perhaps store in compressed format in cache? Lossless PNG can give 2x reduction. Lossy webp up to 5x.
> 
> Downside that it comes at a performance cost when generating thumbs and when accessing a thumb from cache.

performance cost is not a problem if we support only windows that are still supported by Microsoft and other software

at least we are sure that users have good hardware

**Comment by mostlyhuman** [comment]
Hello there, I recently deleted and rebuilt my dynamic thumbnail cache in Icaros and somehow it shrank all the thumbnail sizes when viewed in "Extra Large Icons" mode in windows There used to be a thumbnail view size option in the index location folder area but it isn't there anymore although the 3.3.3.b3 documentation says it should be.  any ideas on this? Thank you for any help!   

Update: I am also having an issue where image thumbnails or at least png's will show a white blank thumbnail, but if I open the image in something like Microsoft Photos then the thumbnail will generate.

**Reply by Xanashi** [reply]
It sounds like you need to rebuild your Windows thumbnail cache too? 
On the Tools page in IcarosConfig, you can clear the Windows cache. Make sure to restart Explorer afterwards to see the effects.

The View Size option was removed on all Windows versions above 8, because all view sizes on newer versions of Windows all use the same sized thumbnails. 

Could you provide a sample of a PNG file that shows this issue?

**Comment by mostlyhuman** [comment]
This was a weird one, I had tried that. I worked on it for hours tried dism
and sfc, did a repair with tweaking.com, finally the only thing that fixed
it was doing an overlay reinstall of windows. Everything is working
perfectly now though, thank you for time I sincerely appreciate it!

**Reply by Xanashi** [reply]
What a weird bug. I'm very glad you found a solution, and thank you so much for sharing it,
so others with the same issue, will know how to fix it.
